### PR TITLE
451006-Sergey Birich

### PR DESCRIPTION
**Pull request by Sergusha-97**

### CODE REVIEW

File: DtoGenerator/DtoGenerator/JsonParser.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.IO;
+ 
+
+namespace DtoGenerator
+{
+    class JsonParser
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/JsonParser.cs]: @Sergusha-97 Explicit access modifier.

File: DtoGenerator/DtoGenerator/PluginLoader.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+namespace DtoGenerator
+{
+    class PluginLoader : Loader
+    {
+        public override List<ITypeMap.ITypeMap> LoadPlugins(string path)
+        {
+            string[] dllFileNames = null;
+            if (Directory.Exists(path))
+            {
+                dllFileNames = Directory.GetFiles(path, "*.dll");
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/PluginLoader.cs]: @Sergusha-97 [Take a look](https://msdn.microsoft.com/en-us/library/system.io.directory.enumeratefiles(v=vs.110).aspx)

File: DtoGenerator/DtoGenerator/Settings.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    class Settings
+    {
+        public readonly int Max_task_number;
+        public readonly string Namespace;
+        public readonly string Destination_folder;
+        public readonly string Source_file;
+        public readonly string Plugins_source_folder;
+        public Settings(NameValueCollection appSettings)
+        {
+            Max_task_number = Convert.ToInt32(appSettings["Max_task_number"]);
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/Settings.cs]: @Sergusha-97 Move this to a builder/factory/etc. and doesn't perform such operations in the constructor. They could take a lot of time.

File: DtoGenerator/DtoGeneratorLib/DtoClassGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            CodeTypeDeclaration classType = new CodeTypeDeclaration(classElem.className);
+            ns.Types.Add(classType);
+            foreach (ClassField elem in classElem.properties)
+            {
+                CreateProperties(elem, compileUnit, classType);
+            }
+            return compileUnit;
+            
+        }
+        private void CreateClass(ClassType classelem)
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGeneratorLib/DtoClassGenerator.cs]: @Sergusha-97 Text file operations should be performed in a separate class.

File: DtoGenerator/DtoGeneratorLib/DtoClassGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                    IndentedTextWriter tw = new IndentedTextWriter(sw, "    ");
+                    provider.GenerateCodeFromCompileUnit(compileUnit, tw,
+                        new CodeGeneratorOptions());
+                    tw.Close();
+                }
+                
+
+                
+        }
+        private void taskFinalization (CountdownEvent countdownEvent)
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGeneratorLib/DtoClassGenerator.cs]: @Sergusha-97 Naming.

File: DtoGenerator/DtoGenerator/Settings.cs (hunk 3 above)
Comment (Sergusha-97) [inline DtoGenerator/DtoGenerator/Settings.cs]: But what will change if I move it to separate method and call it in constructor? or do you offer to call this method after object creation?

File: DtoGenerator/DtoGenerator/PluginLoader.cs (hunk 2 above)
Comment (Sergusha-97) [inline DtoGenerator/DtoGenerator/PluginLoader.cs]: Is it only for familiarization or it's better to use?

File: DtoGenerator/DtoGenerator/JsonParser.cs (hunk 1 above)
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/JsonParser.cs]: @Sergusha-97 Across all your solution, of course.

File: DtoGenerator/DtoGenerator/Settings.cs (hunk 3 above)
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/Settings.cs]: @Sergusha-97 This methods should be called out of the constructor in a separate potentially long running method.

File: DtoGenerator/DtoGenerator/Settings.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Configuration;
+using System.Collections.Specialized;
+namespace DtoGenerator
+{
+   internal class Settings
+    {
+       internal int Max_task_number { get; set; }
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/Settings.cs]: @Sergusha-97 Why have you used such horrible naming?

File: DtoGenerator/DtoGenerator/SettingsBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DtoGenerator
+{
+    internal class SettingsBuilder:Builder
```
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/SettingsBuilder.cs]: @Sergusha-97 A bit strange implementation for such case.

Comment (Sergusha-97) [inline DtoGenerator/DtoGenerator/SettingsBuilder.cs]: I'm implementing this pattern for the first time)))

File: DtoGenerator/DtoGenerator/Settings.cs (hunk 6 above)
Comment (RamanBut-Husaim) [inline DtoGenerator/DtoGenerator/Settings.cs]: @Sergusha-97 `PascalCase` for property naming.

**Pull request merged**